maven-core/src/main/java/org/apache/maven/project/DefaultProjectDependenciesResolver.java

```diff
@@ -187,8 +188,18 @@ public DependencyResolutionResult resolve( DependencyResolutionRequest request )
             {
                 if ( !child.getRelocations().isEmpty() )
                 {
-                    logger.warn( "The artifact " + child.getRelocations().get( 0 ) + " has been relocated to "
-                        + child.getDependency().getArtifact() );
+                    org.eclipse.aether.artifact.Artifact artifact = child.getRelocations().get( 0 );
+                    String message = null;
+                    if ( artifact instanceof RelocatedArtifact )
+                    {
+                        message = ( (RelocatedArtifact) artifact ).getMessage();
+                    }
+                    if ( message == null )
+                    {
+                        message = "The artifact " + artifact + " has been relocated to "
+                                + child.getDependency().getArtifact();
+                    }
```

Comment:
Does relocation message contains coordinate of reallocated artifact from source to destination gav?

If no we lost useful information like:
```
The artifact xxx  has been relocated to yyy
```

Reply:
How about we show the original message and append the contents in `<message>`  if it exists?

Reply:
Correct, "log message: message/reason" otherwise just "log message".